internal/email/email_test.go

```diff
@@ -0,0 +1,160 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package email
+
+import "testing"
+
+func TestIsValidField(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		input          string
+		expectedErr    bool
+		expectedErrMsg string
+	}{
+		// Test case 1: Valid plain text
+		{"Just plain text", false, ""},
+
+		// Test case 2: String with HTML tags
+		{"<b>Bold Text</b>", true, "string <b>Bold Text</b> contains HTML injection"},
+
+		// Test case 3: String with HTML entity
+		{"This is a test &amp; example.", true, "string This is a test &amp; example. contains HTML injection"},
+
+		// Test case 4: String with multiple HTML entities
+		{"This &amp; that &lt; should &gt; work.", true, "string This &amp; that &lt; should &gt; work. contains HTML injection"},
```

Comment:
Looking at this, what prevents us from getting a 1MB string that we try to put into the error message?

(Also, it feels like we should use `%q` here for the text that we're objecting to for clarity.)

Reply:
What do you think is a sane length limitation for these values? Is 200 chars too much?

Reply:
I think 200 characters is probably sufficient.